magnum_cluster_api/resources.py

```diff
@@ -85,6 +86,63 @@ def labels(self) -> dict:
         }
 
 
+class ClusterAutoscalerHelmRepository(Base):
+    def get_object(self) -> objects.HelmRepository:
+        return objects.HelmRepository(
+            self.api,
+            {
+                "apiVersion": objects.HelmRepository.version,
+                "kind": objects.HelmRepository.kind,
+                "metadata": {
+                    "name": "autoscaler",
+                    "namespace": "magnum-system",
+                },
+                "spec": {
+                    "interval": "1m",
+                    "url": "https://kubernetes.github.io/autoscaler",
+                },
+            },
+        )
+
+
+class ClusterAutoscalerHelmRelease(ClusterBase):
+    def get_object(self) -> objects.HelmRelease:
+        cluster_name = utils.get_or_generate_cluster_api_name(self.api, self.cluster)
+        return objects.HelmRelease(
+            self.api,
+            {
+                "apiVersion": objects.HelmRelease.version,
+                "kind": objects.HelmRelease.kind,
+                "metadata": {
+                    "name": f"autoscaler-{self.cluster.uuid}",
```

Comment:
lets use `cluster_name`

Reply:
`cluster_name` is not proper because helmrelease name is used for k8s resource name which requires `DNS RFC1123`. But openstack resource name convention is not following the same rule as k8.
That is why I added `autoscaler-` as the prefix.

Reply:
openstack resource name can be started with number.
But it is not allowed in k8s resource name.

Reply:
aha, ok i found this
magnum resource naming rule is a bit different than others like network
`^[a-zA-Z][a-zA-Z0-9_.-]*$`

Reply:
but k8s service name follows dns-1035
```
The Service "k8s-v1.25.3" is invalid: metadata.name: Invalid value: "k8s-v1.25.3": a DNS-1035 label must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?')
```

Reply:
DNS 1035
```
contain at most 63 characters
contain only lowercase alphanumeric characters or '-'
start with an alphabetic character
end with an alphanumeric character
```

Reply:
@okozachenko1203 can you then make a change to the function that generate cluster name to replace dots with `-` ?

Reply:
let me try